### ISSUE TITLE
Daily Start Date not working correctly for new Dailies when a Custom Day Start is set

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#5395]
I'm noticing on my account with a CDS of 5 that when I create a Daily with a Start Date of today and "Every X Days" greater than 1, that the Daily is not due today. If the number of days is 1, it is due today. My timezone is GMT+10.

The same thing has been reported by another user (987c6d4f-72f2-49a7-888a-e853b11eaff7) with a CDS of 4. GMT+12.

It does not happen on my other account with a CDS of 0 (the Daily is due today when it has a Start Date of today and "Every X Days" greater than or equal to 1). GMT+10.

**Comment by Alys** [comment]
I've added timezones to the description in case they turn out to be relevant (GMT+10 for me, both accounts; GMT+12 for the other user, 987c6d4f-72f2-49a7-888a-e853b11eaff7).

**Comment by Alys** [comment]
Possibly relevant: 

Sometimes when I edit a Daily, the start date appears in this format: `Sat Jun 13 2015 11:55:29 GMT+1000 (EST)` (that was the default start date of a Daily created at exactly that time). That lets me edit the time portion.

For a Daily with Every X Days set to 2, if I change the start date to `Sat Jun 13 2015 04:59:00 GMT+1000 (EST)` (one minute before my 5am CDS time) or any earlier time, the Daily is due today. If I set it to 5am today or after, it is not due.

**Comment by crookedneighbor** [comment]
I think @fallenpanda1's original solution for the weird formatting was to have a temp string variable that gets transformed to a date. I thought I rigged it so that wasn't necessary, but apparently not. Will see about reimplementing it.

**Comment by crookedneighbor** [comment]
For my reference later: https://github.com/HabitRPG/habitrpg/pull/5172/files#diff-6

**Comment by Alys** [comment]
The weird formatting doesn't bother me, and it's useful at the moment for testing for this issue, so I recommend that it's not fixed until later.

**Comment by crookedneighbor** [comment]
I'm saying that the weird formatting may be related to the issue. Sorry, didn't make that clear (sleepy)

**Comment by jylyn** [comment]
I am the other user Alys mentioned above.  I've experimented with setting different start dates with the following results (bearing in mind that today is Jun 13 in my timezone):

Start date Jun 13, repeat every 2 days = grey
Start date Jun 12, repeat every 2 days = due today
Start date Jun 11, repeat every 2 days = grey
Start date Jun 10, repeat every 2 days = due today
Start date Jun 6 or Jun 9, repeat every 3 days = due today, otherwise grey

Basically it seems to me that the script thinks today is Jun 12 even though the display says Jun 13.

**Comment by devbrain83** [comment]
I just noticed the same issue like jylyn and Alys. I think I'll use jylyn's "workaround" until the issue is fixed. Thanks for that.

**Comment by Alys** [comment]
I think I've worked out what's causing this and I'm hoping to create a fix today (might not go live for a couple of days though).

**Comment by crookedneighbor** [comment]
What was it?

**Comment by Alys** [comment]
The everyX code in api.shouldDo isn't correct. It's subtracting a day when it shouldn't be because it's using api.startOfDay incorrectly. I'm doing some small refactoring to make the code clearer, as well as the bug fix. At least that's my current understanding of the problem. Basically I just wanted to say, I'm working on this now so no one else needs to pick it up. :)

**Comment by Alys** [comment]
A fix for this has been created and should be live in the next day or two probably.

**Comment by devbrain83** [comment]
Seems like it's fixed now. Good job!

**Comment by Alys** [comment]
I've made another change that might have fixed any lingering effects from this bug. If anyone still experiences them from their NEXT cron (not from any cron that happened before you read this), please comment here again.
